### PR TITLE
Variable fixes

### PR DESCRIPTION
**Pull request by RoelN**
Small fixes to fully work with actual data from the Wakamai Fondue engine.

### CODE REVIEW

**Comment by RoelN** [comment]
Works in tandem with https://github.com/Wakamai-Fondue/wakamai-fondue-engine/pull/1

**Review by pascalw — APPROVED** [review]
@RoelN if you make changes to the engine that the site depend on, you also have to update the NPM package reference in `package.json` and run `npm update fondue`.

Currently the site depends on `master`, which is resolved to a specific commit when `npm install/update` is ran. We can pick either of two workflows:

1. Always merge changes in the engine first, then update the site PR to use the new commit on master.
2. In site PRs, reference arbitrary commit hashes of yet-to-be-merged changes in the engine.

I think #1 is preferable, as it's easier to reason about. WDYT?

File: src/components/report/VariableControls.js [hunk 1]
```diff
@@ -1,17 +1,11 @@
 export default {
 	props: ["font", "showAxes", "showInstances", "showTitles", "showStyles"],
-	data() {
-		return {
-			activeInstance: ""
-		};
-	},
+	data: self => ({
```
Comment (pascalw) [inline src/components/report/VariableControls.js:3]: I think you can also use `this` in the data function, if you declare it like this:

```js
data() {
  return {
    activeInstance: "",
    axes: self.font.variables.axes,
    instances: self.font.variables.instances
  }
}
```

Comment (RoelN) [reply src/components/report/VariableControls.js:3]: True, I chose this approach to keep in line with the other "data as arrow functions" declarations. Kinda makes it feel consistent, but I'm not married to this approach. Do you prefer the data/return approach?

**Comment by RoelN** [comment]
@pascalw Re: engine/site merge workflow, I prefer #1 as well. Will follow this approach with new PRs!

File: src/components/report/VariableControls.js (hunk 1 above)
Comment (pascalw) [reply src/components/report/VariableControls.js:3]: I'm not sure which style is more typical in Vue. The Vue docs do warn to not use arrow functions in some cases so I think not using arrow functions is a better default.

Screenshot 2020-08-26 at 09 22 56


https://vuejs.org/v2/guide/instance.html#Data-and-Methods

Comment (RoelN) [reply src/components/report/VariableControls.js:3]: Sounds reasonable then! 😄 I'll pick this up in the generic clean up PR I'm working on now.

**Pull request merged**